### ISSUE TITLE
It does not support ts2.8 conditional types ?

### ISSUE THREAD

**Issue by Miloas** [ananthakumaran/tide#245]
```typescript
type foo<T> = T extends Array<infer U> ? U : any
type A = foo<Array<number>> // number
type B = foo<number> // any
```

**Comment by lddubeau** [comment]
I used conditional types in tide *as soon as they came out* and did not run into any problem. (Now, obviously, I had to issue `tide-restart-server` before using conditional types because there's no magic to make tide realize that TypeScript, and consequently, `tsserver` have been upgraded.)  So I'd say tide supports them.

If you post a comment on this issue that details exactly what you expect and what you get instead of what you expect, then we can determine whether you ran into a problem in tide or in `tsserver` (or even with `typescript-mode`, because that's also possible). If the problem is with tide, one of us will reopen this issue. If it is with `tsserver` or `typescript-mode`, we'll direct you to the right place for filing an issue report.
